processing/src/main/java/io/druid/query/aggregation/LongMaxAggregator.java

```diff
@@ -0,0 +1,97 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.query.aggregation;
+
+import io.druid.segment.LongColumnSelector;
+
+import java.util.Comparator;
+
+/**
+ */
+public class LongMaxAggregator implements Aggregator
+{
+  static final Comparator COMPARATOR = LongSumAggregator.COMPARATOR;
+
+  static long combineValues(Object lhs, Object rhs)
+  {
+    return Math.max(((Number) lhs).longValue(), ((Number) rhs).longValue());
+  }
+
+  private final LongColumnSelector selector;
+  private final String name;
+
+  private long max;
```

Comment:
none of the other existing aggregators don't have it as volatile. i think even final value also gets called on the same thread.
however, if we do decide to change it to volatile then we need to do it for all the aggregators in a separate PR.